shared/teams/team/header/container.js

```diff
@@ -36,6 +37,10 @@ const mapDispatchToProps = (dispatch: Dispatch, {teamname}: OwnProps) => ({
   },
   onChat: () => dispatch(Chat2Gen.createPreviewConversation({teamname, reason: 'teamHeader'})),
   onEditDescription: () => dispatch(navigateAppend([{props: {teamname}, selected: 'editTeamDescription'}])),
+  onEditIcon: (image?: Response) =>
+    dispatch(
+      navigateAppend([{props: {image, sendChatNotification: true, teamname}, selected: 'editTeamAvatar'}])
```

Comment:
do we really have to use routeprops for this

Reply:
How else can I do it?

Reply:
it could go into the store. i think this is fine for now as this PR is too big and too old but as a future reference i think we should avoid routeProps unless absolutely necessary (aka when a single screen is used multiple places and needs context passed in). instead probably simpler to just use the store

Reply:
Sounds good. Thanks for the info!